sstables_loader.cc

```diff
@@ -502,13 +525,30 @@ class sstables_loader::download_task_impl : public tasks::task_manager::task::im
     }
 
     virtual future<tasks::task_manager::task::progress> get_progress() const override {
-        llog.debug("get_progress: {}", _num_sstables_processed);
-        unsigned processed = co_await _loader.map_reduce(adder<unsigned>(), [this] (auto&) {
-            return _num_sstables_processed[this_shard_id()];
-        });
+        struct adder {
+            progress result;
+            future<> operator()(const progress p) {
+                llog.debug("get_progress: {} / {}", p.processed, p.total);
+                result.processed += p.processed;
+                if (result.total == 0.) {
+                    // total number of batches can be tracked on multiple shards,
+                    // let's keep the first non-zero one.
+                    result.total = p.total;
```

Comment:
Why is that so? E.g. the `sstable_streamer::stream()` calls progress update like `on_stream(0, _sstables.size())` which means that local progress' .total is updated with local _sstables.size(), respectively total number of sstables should be summed on all shards, not just picked from "random" one

Reply:
thank you for pointing this out indeed, we split the sstables by shard, so we should sum them up, instead of picking a "random" one.